### ISSUE TITLE
HomeConnect integration does not handle dishwasher program changes correctly

### ISSUE THREAD

**Issue by RikvdMark** [home-assistant/home-assistant.github.io#33749]
### Feedback

Hi,

First off a big thank you for this Integration. I didn't expect much and just wondered if I could monitor my dishwasher from my Home Assisant.

It works and does that however I have noticed one odity,
When I link my HomeConnect application it perfectly adds my dishwasher and the program it is running at the time. But no entities are created in HomeAssistant for the other programs. So if I initially added the application/device during a "auto" cleaning cycle a "switch.dishbosch_program_auto2" is created because that program is running.
But if I then run the "eco" program next time it only show that the dishwasher is running but since there is no entity for the "auto" program the HomeConnect integration cannot show that in Home Assistant.

If I then remove the device/application application from the HomeConnect integration and link it again a entity "switch.dishbosch_program_eco50" is created because that program is running at the time.
Of course then the "switch.dishbosch_program_auto2" entiy no longer exists so my dashboard logically complains and next time I run the "auto" program Home Assistant again shows the dishwasher is running but not that it is running auto.

I have included a few screenshots for clarification:
![Screenshot 2024-07-14 152018](https://github.com/user-attachments/assets/7c821b67-647d-4673-896c-bda6d12b9b24)
![Screenshot 2024-07-14 151826](https://github.com/user-attachments/assets/b56ef31c-3c18-4c9c-89da-19ffa36cec53)
![Screenshot 2024-07-14 151926](https://github.com/user-attachments/assets/08a466ca-c816-4606-a0c4-616267461614)

I do not expect a fix but it would be nice if it could be fixed :)

Thanks in advance and keep up the good work!

Kind Regards,
Rik van der Mark

### URL

https://www.home-assistant.io/integrations/home_connect/

### Version

2024.7.2

**Comment by peterkh** [comment]
Just set this up for my Bosch washing machine and seeing a similar behaviour in terms of program entities.

**Comment by GSzabados** [comment]
I want to highlight for both of you this section of the documentation:

Important:

- Power on all your appliances during the integration configuration process; otherwise appliance programs list will be empty.

**- To update the appliance programs list, you can reload the Home Connect integration when an appliance is turned on. If the re-initialization process is not triggered by reload, restart the Home Assistant when an appliance is turned on.**

- All changes in the developer portal take 15 minutes before the change is implemented.

To get a full program list, turn on your device, do not start it, just turn it on and then reload your integration in HA.

**Comment by peterkh** [comment]
thanks @GSzabados missed that last line in the docs, which is what fixed it for me.

**Comment by GSzabados** [comment]
@peterkh, if you meant this, then that was my consolidated instruction in layman-terms. 😆 

> To get a full program list, turn on your device, do not start it, just turn it on and then reload your integration in HA.

**Comment by RikvdMark** [comment]
Thank you! @GSzabados
I see where I went wrong, I integrated it at first when it was on AND running a program.
For some reason it did get all the entities later shortly after receiving a software update. It's a nice to have to monitor a dishwasher from HA but I'm glad it's all working now.

![Home Assistant - DishBosh - working](https://github.com/user-attachments/assets/f7562d9d-43e3-4267-a3c0-bfd7fda24e54)
